compiler_opt/rl/generate_bc_trajectories.py

```diff
@@ -809,3 +827,162 @@ def select_best_exploration(
         module_dict_max,
         module_dict_pol,
     ), seq_example.SerializeToString()
+
+
+@gin.configurable
+def gen_trajectories(
+    data_path: str = gin.REQUIRED,
+    delete_flags: Tuple[str, ...] = gin.REQUIRED,
+    output_file_name: str = gin.REQUIRED,
+    output_path: str = gin.REQUIRED,
+    obs_action_spec: Optional[Tuple[time_step.TimeStep,
+                                    tensor_spec.BoundedTensorSpec]] = None,
+    num_workers: Optional[int] = None,
+    num_output_files: int = 1,
+    profiling_file_path: Optional[str] = None,
+    worker_wait: int = 10,
+    worker_manager_class=local_worker_manager.LocalWorkerPoolManager,
+):
+  """Generates all trajectories for imitation learning training.
+
+  Args:
+    data_path: path to the corpus of modules.
+    delete_flags: flags to be deleted during compilation.
+    output_file_name: name of the files to write to.
+    output_path: path to save the files to.
+    time_step_spec: optional observation spec annotating the
+      state (TimeStep) for training a policy
+    obs_action_spec: optional observation and action spec annotating the state
+      (TimeStep) for training a policy
+    num_workers: number of distributed workers to process the corpus.
+    num_output_files: number of files to partition the outputs into, if set to n
+      then each file is names output_file_name-i-of-n.tfrecord
+    worker_wait: max number of seconds to wait for a worker to terminate
+    worker_manager_class: A pool of workers hosted on the local machines, each
+      in its own process.
+  """
+  cps = corpus.Corpus(data_path=data_path, delete_flags=delete_flags)
+  logging.info('Done loading module specs from corpus.')
+
+  corpus_elements = cps.module_specs
+  work = [
+      cps.load_module_spec(corpus_element) for corpus_element in corpus_elements
+  ]
+
+  modules_processed = 0
+  time_compiler_calls = 0
+  total_successful_examples = 0
+  total_work = len(corpus_elements)
+  total_failed_examples = 0
+  total_write_files = num_output_files
+  total_profiles_max: List[Optional[Dict[str, Union[str, float, int]]]] = []
```

Comment:
there is a lot of nesting. A trick to help comprehension is to alias the type somewhere (as a module-level def), e.g.: (I'm making the names up)

```
ExperimentValueType = Union[str, float, int]
ExperimentResultType = Dict[str, ExperimentValueType]
```